Documentation/Evolution/UnicodeForStringProcessing.md

```diff
@@ -0,0 +1,691 @@
+
+| Matching  `"Cafe\u{301}"` | Pattern: `^Caf.` | Remaining | Pattern:  `^Caf\X` | Remaining |
+|---|---|---|---|---|
+| C#, Rust, Go | `"Cafe"` | `"´"` | n/a | n/a |
+| NSString, Java, Ruby, Perl | `"Cafe"` | `"´"` | `"Café"` | `""` |
+
+Other than Java's `CANON_EQ` option, the vast majority of other languages and engines are not capable of comparing with canonical equivalence.
+
+</details>
+
+## Proposed solution
+
+TK: semantic levels, options for controlling, canonical equivalence, Unicode properties
+
+## Detailed design
+
+First, we'll discuss the options that let you control a regex's behavior, and then explore the character classes that define the your pattern.
+
+### Options
+
+Options can be enabled and disabled in two different ways: as part of [regular expression internal syntax][internals], or applied as methods when declaring a `Regex`. For example, both of these `Regex`es are declared with case insensitivity:
+
+```swift
+let regex1 = /(?i)banana/
+let regex2 = Regex {
+    "banana"
+}.ignoringCase()`
+```
+
+Note that the `ignoringCase()` is available on any type conforming to `RegexComponent`, which means that you can use the more readable option-setting interface in conjunction with regex literals:
+
+```swift
+let regex3 = /banana/.ignoringCase()
+```
+
+Calling an option-setting method like `ignoringCase(_:)` acts like wrapping the callee in an option-setting group `(?:...)`. That is, while it sets the behavior for the callee, it doesn’t override options that are applied to more specific regions. In this example, the middle `"na"` in `"banana"` matches case-sensitively, despite the outer call to `ignoringCase()`:
+
+```swift
+let regex4 = Regex {
+    "ba"
+    "na".ignoringCase(false)
+    "na"
+}
+.ignoringCase()
+	
+"banana".contains(regex4)     // true
+"BAnaNA".contains(regex4)     // true
+"BANANA".contains(regex4)     // false
+
+// Equivalent to:
+let regex5 = /(?i)ba(?-i:na)na/
+```
+
+#### Case insensitivity
+
+Regular expressions perform case sensitive comparisons by default. The `i` option or the `ignoringCase(_:)` method enables case insensitive comparison.
+
+```swift
+let str = "Café"
+	
+str.firstMatch(of: /CAFÉ/)          // nil
+str.firstMatch(of: /(?i)CAFÉ/)      // "Café"
+str.firstMatch(of: /(?i)cAfÉ/)      // "Café"
+```
+
+Case insensitive matching uses case folding to ensure that canonical equivalence continues to operate as expected.
+
+**Regular expression syntax:** `(?i)...` or `(?i:...)`
+
+**`RegexBuilder` API:**
+
+```swift
+extension RegexComponent {
+    /// Returns a regular expression that ignores casing when matching.
+    public func ignoringCase(_ ignoreCase: Bool = true) -> Regex<Output>
+}
+```
+
```

Comment:
These are defined on `RegexComponent` instead of `Regex`, so `ignoringCase()` is available on `String` whenever `String` conforms. Unfortunately, it looks like that conformance is in `_StringProcessing` instead of `RegexBuilder`! I'll move it over and explain the way it works in the proposal.